pkg/cmd/run/watch/watch.go

```diff
@@ -0,0 +1,228 @@
+package watch
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"runtime"
+	"time"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/run/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+const defaultInterval int = 3
+
+type WatchOptions struct {
+	IO         *iostreams.IOStreams
+	HttpClient func() (*http.Client, error)
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	RunID      string
+	Interval   int
+	ExitStatus bool
+
+	Prompt bool
+
+	Now func() time.Time
+}
+
+func NewCmdWatch(f *cmdutil.Factory, runF func(*WatchOptions) error) *cobra.Command {
+	opts := &WatchOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Now:        time.Now,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "watch <run-selector>",
+		Short: "Runs until a run completes, showing its progress",
```

Comment:
I think "Watch a run until it completes, showing its progress" is a bit less confusing.